src/AnnotationResolver.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Opstalent\CrudBundle;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\ORM\Mapping\Column;
+use Opstalent\CrudBundle\Annotation\Entity;
+use Opstalent\CrudBundle\Annotation\Field;
+use Opstalent\CrudBundle\Exception\AnnotationNotDefinedException;
+use Opstalent\CrudBundle\Exception\ClassNotFoundException;
+use ReflectionClass;
+use ReflectionProperty;
+
+/**
+ * Class AnnotationResolver
+ * @author Szymon Kunowski <szymon.kunowski@gmail.com>
+ * @package Opstalent\CrudBundle
+ */
+class AnnotationResolver
+{
+    /**
+     * @var AnnotationReader
+     */
+    protected static $reader;
+
+    /**
+     * @param string $action
+     * @param string $className
+     * @return array
+     */
+    public static function resolve(string $action, string $className): array
+    {
+        $reflection = static::getReflectionClass($className);
+        if (!static::isEntity($reflection, $action)) {
+            throw new AnnotationNotDefinedException();
+        }
+        $properties = [];
+        foreach ($reflection->getProperties() as $property) {
+            $field = static::getReader()->getPropertyAnnotation($property, Field::class);
+            if (!$field || !$field->isValidAction($action)) {
+                continue;
+            }
+
+            $column = static::getColumnAnnotation($property);
+            $properties[$property->getName()] = $column->type;
```

Comment:
why not `$properties[$property->getName()] = $this->resolveProperty($property);`?